{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"species_NCBI.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "links = df[['GenBank FTP']].values\n",
    "names = df[['#Organism Name']].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "link = str(links[0][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/004/010/195/GCA_004010195.1_ASM401019v1'"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "link"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "download file: chr1.fna.gz\n",
      "download file: chr10.fna.gz\n",
      "download file: chr11.fna.gz\n",
      "download file: chr12.fna.gz\n",
      "download file: chr13.fna.gz\n",
      "download file: chr14.fna.gz\n",
      "download file: chr15.fna.gz\n",
      "download file: chr16.fna.gz\n",
      "download file: chr17.fna.gz\n",
      "download file: chr18.fna.gz\n",
      "download file: chr19.fna.gz\n",
      "download file: chr2.fna.gz\n",
      "download file: chr20.fna.gz\n",
      "download file: chr21.fna.gz\n",
      "download file: chr22.fna.gz\n",
      "download file: chr23.fna.gz\n",
      "download file: chr24.fna.gz\n",
      "download file: chr25.fna.gz\n",
      "download file: chr26.fna.gz\n",
      "download file: chr27.fna.gz\n",
      "download file: chr28.fna.gz\n",
      "download file: chr29.fna.gz\n",
      "download file: chr3.fna.gz\n",
      "download file: chr30.fna.gz\n",
      "download file: chr31.fna.gz\n",
      "download file: chr32.fna.gz\n",
      "download file: chr33.fna.gz\n",
      "download file: chr34.fna.gz\n",
      "download file: chr35.fna.gz\n",
      "download file: chr36.fna.gz\n",
      "download file: chr37.fna.gz\n",
      "download file: chr38.fna.gz\n",
      "download file: chr39.fna.gz\n",
      "download file: chr4.fna.gz\n",
      "download file: chr40.fna.gz\n",
      "download file: chr41.fna.gz\n",
      "download file: chr42.fna.gz\n",
      "download file: chr43.fna.gz\n",
      "download file: chr44.fna.gz\n",
      "download file: chr45.fna.gz\n",
      "download file: chr46.fna.gz\n",
      "download file: chr47.fna.gz\n",
      "download file: chr48.fna.gz\n",
      "download file: chr49.fna.gz\n",
      "download file: chr5.fna.gz\n",
      "download file: chr50.fna.gz\n",
      "download file: chr51.fna.gz\n",
      "download file: chr6.fna.gz\n",
      "download file: chr7.fna.gz\n",
      "download file: chr8.fna.gz\n",
      "download file: chr9.fna.gz\n",
      "download file: chr1.fna.gz\n",
      "download file: chr10.fna.gz\n",
      "download file: chr11.fna.gz\n",
      "download file: chr12.fna.gz\n",
      "download file: chr13.fna.gz\n",
      "download file: chr14.fna.gz\n",
      "download file: chr15.fna.gz\n",
      "download file: chr16.fna.gz\n",
      "download file: chr17.fna.gz\n",
      "download file: chr18.fna.gz\n",
      "download file: chr19.fna.gz\n",
      "download file: chr2.fna.gz\n",
      "download file: chr20.fna.gz\n",
      "download file: chr21.fna.gz\n",
      "download file: chr22.fna.gz\n",
      "download file: chr23.fna.gz\n",
      "download file: chr24.fna.gz\n",
      "download file: chr25.fna.gz\n",
      "download file: chr26.fna.gz\n",
      "download file: chr27.fna.gz\n",
      "download file: chr28.fna.gz\n",
      "download file: chr29.fna.gz\n",
      "download file: chr3.fna.gz\n",
      "download file: chr30.fna.gz\n",
      "download file: chr31.fna.gz\n",
      "download file: chr32.fna.gz\n",
      "download file: chr33.fna.gz\n",
      "download file: chr34.fna.gz\n",
      "download file: chr35.fna.gz\n",
      "download file: chr36.fna.gz\n",
      "download file: chr37.fna.gz\n",
      "download file: chr38.fna.gz\n",
      "download file: chr39.fna.gz\n",
      "download file: chr4.fna.gz\n",
      "download file: chr40.fna.gz\n",
      "download file: chr41.fna.gz\n",
      "download file: chr42.fna.gz\n",
      "download file: chr43.fna.gz\n",
      "download file: chr44.fna.gz\n",
      "download file: chr45.fna.gz\n",
      "download file: chr46.fna.gz\n",
      "download file: chr5.fna.gz\n",
      "download file: chr6.fna.gz\n",
      "download file: chr7.fna.gz\n",
      "download file: chr8.fna.gz\n",
      "download file: chr9.fna.gz\n",
      "download file: chr1.fna.gz\n",
      "download file: chr10.fna.gz\n",
      "download file: chr11.fna.gz\n",
      "download file: chr12.fna.gz\n",
      "download file: chr13.fna.gz\n",
      "download file: chr14.fna.gz\n",
      "download file: chr15.fna.gz\n",
      "download file: chr16.fna.gz\n",
      "download file: chr17.fna.gz\n",
      "download file: chr18.fna.gz\n",
      "download file: chr19.fna.gz\n",
      "download file: chr2.fna.gz\n",
      "download file: chr20.fna.gz\n",
      "download file: chr21.fna.gz\n",
      "download file: chr22.fna.gz\n",
      "download file: chr23.fna.gz\n",
      "download file: chr24.fna.gz\n",
      "download file: chr25.fna.gz\n",
      "download file: chr26.fna.gz\n",
      "download file: chr27.fna.gz\n",
      "download file: chr28.fna.gz\n",
      "download file: chr29.fna.gz\n",
      "download file: chr3.fna.gz\n",
      "download file: chr30.fna.gz\n",
      "download file: chr31.fna.gz\n",
      "download file: chr32.fna.gz\n",
      "download file: chr33.fna.gz\n",
      "download file: chr34.fna.gz\n",
      "download file: chr35.fna.gz\n",
      "download file: chr36.fna.gz\n",
      "download file: chr37.fna.gz\n",
      "download file: chr38.fna.gz\n",
      "download file: chr39.fna.gz\n",
      "download file: chr4.fna.gz\n",
      "download file: chr40.fna.gz\n",
      "download file: chr41.fna.gz\n",
      "download file: chr42.fna.gz\n",
      "download file: chr43.fna.gz\n",
      "download file: chr44.fna.gz\n",
      "download file: chr45.fna.gz\n",
      "download file: chr46.fna.gz\n",
      "download file: chr47.fna.gz\n",
      "download file: chr48.fna.gz\n",
      "download file: chr49.fna.gz\n",
      "download file: chr5.fna.gz\n",
      "download file: chr6.fna.gz\n",
      "download file: chr7.fna.gz\n",
      "download file: chr8.fna.gz\n",
      "download file: chr9.fna.gz\n",
      "download file: chr1.fna.gz\n",
      "download file: chr10.fna.gz\n",
      "download file: chr11.fna.gz\n",
      "download file: chr12.fna.gz\n",
      "download file: chr13.fna.gz\n",
      "download file: chr14.fna.gz\n",
      "download file: chr15.fna.gz\n",
      "download file: chr16.fna.gz\n",
      "download file: chr17.fna.gz\n",
      "download file: chr18.fna.gz\n",
      "download file: chr19.fna.gz\n",
      "download file: chr2.fna.gz\n",
      "download file: chr20.fna.gz\n",
      "download file: chr21.fna.gz\n",
      "download file: chr22.fna.gz\n",
      "download file: chr23.fna.gz\n",
      "download file: chr24.fna.gz\n",
      "download file: chr25.fna.gz\n",
      "download file: chr26.fna.gz\n",
      "download file: chr27.fna.gz\n",
      "download file: chr28.fna.gz\n",
      "download file: chr29.fna.gz\n",
      "download file: chr3.fna.gz\n",
      "download file: chr30.fna.gz\n",
      "download file: chr31.fna.gz\n",
      "download file: chr32.fna.gz\n",
      "download file: chr33.fna.gz\n",
      "download file: chr34.fna.gz\n",
      "download file: chr35.fna.gz\n",
      "download file: chr36.fna.gz\n",
      "download file: chr37.fna.gz\n",
      "download file: chr38.fna.gz\n",
      "download file: chr39.fna.gz\n",
      "download file: chr4.fna.gz\n",
      "download file: chr40.fna.gz\n",
      "download file: chr41.fna.gz\n",
      "download file: chr42.fna.gz\n",
      "download file: chr43.fna.gz\n",
      "download file: chr44.fna.gz\n",
      "download file: chr45.fna.gz\n",
      "download file: chr46.fna.gz\n",
      "download file: chr47.fna.gz\n",
      "download file: chr48.fna.gz\n",
      "download file: chr49.fna.gz\n",
      "download file: chr5.fna.gz\n",
      "download file: chr50.fna.gz\n",
      "download file: chr51.fna.gz\n",
      "download file: chr52.fna.gz\n",
      "download file: chr53.fna.gz\n",
      "download file: chr54.fna.gz\n",
      "download file: chr55.fna.gz\n",
      "download file: chr56.fna.gz\n",
      "download file: chr57.fna.gz\n",
      "download file: chr58.fna.gz\n",
      "download file: chr59.fna.gz\n",
      "download file: chr6.fna.gz\n",
      "download file: chr60.fna.gz\n",
      "download file: chr7.fna.gz\n",
      "download file: chr8.fna.gz\n",
      "download file: chr9.fna.gz\n",
      "download file: chr1.fna.gz\n",
      "download file: chr10.fna.gz\n",
      "download file: chr11.fna.gz\n",
      "download file: chr12.fna.gz\n",
      "download file: chr13.fna.gz\n",
      "download file: chr14.fna.gz\n",
      "download file: chr15.fna.gz\n",
      "download file: chr16.fna.gz\n",
      "download file: chr17.fna.gz\n",
      "download file: chr18.fna.gz\n",
      "download file: chr19.fna.gz\n",
      "download file: chr2.fna.gz\n",
      "download file: chr20.fna.gz\n",
      "download file: chr21.fna.gz\n",
      "download file: chr22.fna.gz\n",
      "download file: chr23.fna.gz\n",
      "download file: chr3.fna.gz\n",
      "download file: chr4.fna.gz\n",
      "download file: chr5.fna.gz\n",
      "download file: chr6.fna.gz\n",
      "download file: chr7.fna.gz\n",
      "download file: chr8.fna.gz\n",
      "download file: chr9.fna.gz\n"
     ]
    }
   ],
   "source": [
    "from ftplib import FTP\n",
    "import os, gzip\n",
    "\n",
    "files = {}\n",
    "paths = {}\n",
    "\n",
    "def files_callback(name):\n",
    "    files_array.append(name)\n",
    "\n",
    "for index in range(len(links)):\n",
    "    link = links[index][0]\n",
    "    parts = link.split('/', maxsplit=3)\n",
    "    \n",
    "    next_path = (parts[3].split('/')[-1] + '_assembly_structure/') + 'Primary_Assembly/assembled_chromosomes/FASTA/'\n",
    "    ftp = FTP(parts[2])\n",
    "    ftp.login()\n",
    "    ftp.cwd(parts[3] + '/' + next_path)\n",
    "    paths[names[index][0]] = parts[3] + '/' + next_path\n",
    "    #ftp.cwd('genomes/all/GCA/902/167/405/GCA_902167405.1_gadMor3.0/GCA_902167405.1_gadMor3.0_assembly_structure/Primary_Assembly/assembled_chromosomes/FASTA/')\n",
    "    files_array = []\n",
    "    ftp.retrlines('NLST', callback = files_callback)\n",
    "    files[names[index][0]] = files_array\n",
    "    ftp.close()\n",
    "    \n",
    "for name, path in paths.items():\n",
    "    server = 'ftp.ncbi.nlm.nih.gov'\n",
    "    ftp = FTP(server)\n",
    "    ftp.login()\n",
    "    ftp.cwd(path)\n",
    "    os.mkdir(name)\n",
    "    os.mkdir(name + '/' + 'dna_sm')\n",
    "    for filename in files[name]:\n",
    "        print('download file: ' + filename)\n",
    "        file_path = os.path.join(name, 'dna_sm', filename)\n",
    "        fasta_path = os.path.join(name, 'dna_sm', filename[0:-3])\n",
    "        with open(file_path, 'wb') as file:\n",
    "            ftp.retrbinary('RETR ' + filename, file.write, 102400)\n",
    "        handle = gzip.open(file_path)\n",
    "        with open(fasta_path, 'wb') as out:\n",
    "            for line in handle:\n",
    "                out.write(line)\n",
    "    ftp.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'250 CWD command successful'"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ftp.cwd('genomes/all/GCA/902/167/405/GCA_902167405.1_gadMor3.0/GCA_902167405.1_gadMor3.0_assembly_structure/Primary_Assembly/assembled_chromosomes/FASTA/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Chiloscyllium plagiosum': 'genomes/all/GCA/004/010/195/GCA_004010195.1_ASM401019v1/GCA_004010195.1_ASM401019v1_assembly_structure/Primary_Assembly/assembled_chromosomes/FASTA/',\n",
       " 'Pristis pectinata': 'genomes/all/GCA/009/764/475/GCA_009764475.1_sPriPec2.pri/GCA_009764475.1_sPriPec2.pri_assembly_structure/Primary_Assembly/assembled_chromosomes/FASTA/',\n",
       " 'Amblyraja radiata': 'genomes/all/GCA/010/909/765/GCA_010909765.1_sAmbRad1.pri/GCA_010909765.1_sAmbRad1.pri_assembly_structure/Primary_Assembly/assembled_chromosomes/FASTA/',\n",
       " 'Acipenser ruthenus': 'genomes/all/GCA/010/645/085/GCA_010645085.1_ASM1064508v1/GCA_010645085.1_ASM1064508v1_assembly_structure/Primary_Assembly/assembled_chromosomes/FASTA/',\n",
       " 'Gadus morhua': 'genomes/all/GCA/902/167/405/GCA_902167405.1_gadMor3.0/GCA_902167405.1_gadMor3.0_assembly_structure/Primary_Assembly/assembled_chromosomes/FASTA/'}"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.1.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.10.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.11.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.12.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.13.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.14.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.15.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.16.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.17.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.18.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.19.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.2.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.20.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.21.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.22.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.23.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.24.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.25.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.26.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.27.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.28.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.3.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.4.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.5.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.6.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.7.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.8.fa\n",
      "Analysing: Rhinolophus_ferrumequinum/dna_sm/Rhinolophus_ferrumequinum.mRhiFer1_v1.p.dna_sm.primary_assembly.9.fa\n",
      "creating jsons/Rhinolophus_ferrumequinum.json\n"
     ]
    }
   ],
   "source": [
    "from dna_puller.sliding_parser import SlidingParser\n",
    "import os, json\n",
    "\n",
    "names = [ 'Rhinolophus_ferrumequinum'\n",
    "]\n",
    "\n",
    "for index in range(len(names)):\n",
    "    name = names[index]\n",
    "    path = name + \"/dna_sm\"\n",
    "    data = {'dna_sm': {}}\n",
    "    \n",
    "    files = os.listdir(path)\n",
    "    fasta_files = []\n",
    "    \n",
    "    for file in files:\n",
    "        if file[-3:] == '.fa':\n",
    "            fasta_files.append(file)\n",
    "    \n",
    "    for file in fasta_files:\n",
    "        parser = SlidingParser(1000)\n",
    "        data['dna_sm'].update(parser.parse_file(path + \"/\" + file))\n",
    "    \n",
    "    json_path = os.path.join('jsons' , name + '.json')\n",
    "    print('creating ' + json_path)\n",
    "    with open(json_path, 'w') as fp:\n",
    "         json.dump(data, fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "json_files = ['Pristis_pectinata.json',\n",
    "'Chiloscyllium_plagiosum.json',\n",
    "#'Myotis_lucifugus.json',\n",
    "#'Pteropus_vampyrus.json',\n",
    "#'Rhinolophus_ferrumequinum.json'\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv, os, json\n",
    "\n",
    "file_counts = {}\n",
    "\n",
    "\n",
    "for file in json_files:\n",
    "    name = file[:-5]\n",
    "    data = {}\n",
    "    if not os.path.exists('csv/' + name):\n",
    "        os.mkdir('csv/' + name)\n",
    "    with open('jsons/' + file) as f:\n",
    "        data = json.load(f)\n",
    "    file_counts[file] = {'sm': [], 'all': []}\n",
    "    for lg, values in data.items():\n",
    "        if lg[:2] == 'LR' or lg[:2] == 'CM':\n",
    "            with open('csv/' + name + '/' + lg + '.csv', 'w', newline='') as csvfile:\n",
    "                #print(lg)\n",
    "                csvwriter = csv.writer(csvfile, delimiter=' ',\n",
    "                                        quotechar='|', quoting=csv.QUOTE_MINIMAL)\n",
    "\n",
    "                for index, value in values.items():\n",
    "                    gc_count = value['G'] + value['C'] + value['S'] + value['g'] + value['c'] + value['s']\n",
    "                    all_count = value['all'] - value['N'] - value['n']\n",
    "\n",
    "                    big_percent = 0 \n",
    "                    if value['all_big'] > 0:\n",
    "                        big_percent = value['all_big'] / float(value['all'])\n",
    "                        if big_percent < 1:\n",
    "                            file_counts[file]['sm'].append(big_percent)\n",
    "                        file_counts[file]['all'].append(big_percent)\n",
    "\n",
    "                    if all_count > 0:\n",
    "                        gc_val = gc_count/all_count\n",
    "\n",
    "                    csvwriter.writerow([index, gc_val, big_percent])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "331"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(file_counts['gadus_morhua.json']['sm'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os, subprocess\n",
    "\n",
    "def file_len(fname):\n",
    "    p = subprocess.Popen(['wc', '-l', fname], stdout=subprocess.PIPE, \n",
    "                                              stderr=subprocess.PIPE)\n",
    "    result, err = p.communicate()\n",
    "    if p.returncode != 0:\n",
    "        raise IOError(err)\n",
    "    return int(result.strip().split()[0])\n",
    "\n",
    "csv_folders = 'csv'\n",
    "\n",
    "#folders = os.listdir(csv_folders)\n",
    "folders = ['Oryzias_latipes',\n",
    "'Esox_lucius',\n",
    "'Betta_splendens',\n",
    "'Lepisosteus_oculatus',\n",
    "'Danio_rerio',\n",
    "'Erpetoichthys_calabaricus',\n",
    "'Salmo_salar',\n",
    "'Takifugu_rubripes',\n",
    "#'Acipenser_ruthenus'\n",
    "'Ciona_intestinalis',\n",
    "'Felis_catus',\n",
    "'Gorilla_gorilla'\n",
    "]\n",
    "\n",
    "folders = ['Pristis_pectinata',\n",
    "'Chiloscyllium_plagiosum']\n",
    "\n",
    "species_folders = []\n",
    "\n",
    "for folder in folders:\n",
    "    if folder[0] != '.':\n",
    "        species_folders.append(folder)\n",
    "\n",
    "files = {}\n",
    "biggest_files = {}\n",
    "sorted_files = {}\n",
    "for folder in species_folders:\n",
    "    biggest_files[folder] = 0\n",
    "    sorted_files[folder] = {}\n",
    "    csv_files = []\n",
    "    for file in os.listdir(csv_folders + '/' + folder):\n",
    "        if file[-4:] == '.csv':\n",
    "            csv_files.append(file)\n",
    "            \n",
    "            len_f =  file_len(csv_folders + '/' + folder + '/' + file)\n",
    "            sorted_files[folder][file] = len_f\n",
    "            if biggest_files[folder] < len_f * 1000:\n",
    "                biggest_files[folder] = len_f * 1000\n",
    "    files[folder] = csv_files \n",
    "    sorted_files[folder] = sorted(sorted_files[folder], key = sorted_files[folder].get, reverse=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gnuplot_generator.gnuplot_generator import GnuplotGenerator\n",
    "\n",
    "def create_plot(fname):\n",
    "    p = subprocess.Popen(['gnuplot', '-p', fname], stdout=subprocess.PIPE, \n",
    "                                                   stderr=subprocess.PIPE)\n",
    "    result, err = p.communicate()\n",
    "    if p.returncode != 0:\n",
    "        raise IOError(err)       \n",
    "postfix = 'normal'    \n",
    "        \n",
    "for folder in species_folders:\n",
    "    range_plot = '[0:' + str(biggest_files[folder]) + ']'\n",
    "    gen = GnuplotGenerator(folder, range_plot)\n",
    "    gen.add_palette(\"( 0 'green', 50 'orange', 100 'red')\")\n",
    "    gen.set_term('png', 'plots_' + postfix + '/' + folder + '_' + postfix + '.png', 10000, 0.1)\n",
    "    \n",
    "    if not os.path.exists('plots_' + postfix):\n",
    "        os.makedirs('plots_' + postfix)\n",
    "    \n",
    "    for file in sorted_files[folder]:\n",
    "        gen.add_plot(csv_folders + '/' + folder + '/' + file, '1:2:3', file[:-4])\n",
    "    \n",
    "    with open(csv_folders + '/' + folder + '/' + folder + '_' + postfix + '.gnu', 'w') as gnu_file:\n",
    "        lines = gen.prepare_definition()\n",
    "        for line in lines:\n",
    "            gnu_file.write(line + '\\n')\n",
    "    \n",
    "    create_plot(csv_folders + '/' + folder + '/' + folder + '_' + postfix + '.gnu')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
